src/main/java/xbot/common/trajectory/SwerveSimpleTrajectoryLogic.java

```diff
@@ -263,6 +295,109 @@ private List<XbotSwervePoint> getVelocityAdjustedSwervePoints(
         return velocityAdjustedPoints;
     }
 
+    private List<XbotSwervePoint> getKinematicsAdjustedSwervePoints(
+            XbotSwervePoint initialPoint,
+            List<XbotSwervePoint> swervePoints) {
+
+        ArrayList<XbotSwervePoint> adjustedPoints = new ArrayList<>();
+
+        SwerveKinematicsCalculator calculator = null;
+        for (int i = 0; i < swervePoints.size(); i++) {
+            XbotSwervePoint previous = initialPoint;
```

Comment:
⭐ ⭐ this is a little awkward to have inside the loop. Instead consider setting this above before the loop and then at the bottom of the loop assign previous to the current to update it